### PR TITLE
Tempest passed - Elk configurations

### PR DESCRIPTION
**Pull request by genagain**
This pull request installs the filebeat service on the compute and controller nodes. Once filebeat is running on a node, the any of the log files can be forwarded to an server running the ELK stack. Once the logs from different nodes are being aggregated by the server running the ELK stack, log files can be searched and visualized really quickly

### CODE REVIEW

**Comment by genagain** [comment]
I ran this in my staging environment and it worked like expected.

**Comment by genagain** [comment]
The following parameter needs to be added to the hiera file:

```
quickstack::params::elasticsearch_host: elk_stack_vm_ip:5044
```

**Comment by genagain** [comment]
Tempest worked the way it was supposed to!

File: quickstack/manifests/controller_common.pp [hunk 1]
```diff
@@ -840,6 +840,22 @@
     cron_hour      => $backups_hour,
     cron_min       => $backups_min,
   }
+  
+  class { 'filebeat':
+    outputs => {
+      'logstash'  => {
+      'hosts'        =>  [$elasticsearch_host],
```
Comment (lokI8) [inline quickstack/manifests/controller_common.pp]: can you indent this and the next line once more for clarity?

and i don't see the elasticsearch_host parameter to the class definition at the top like in compute_common.

**Comment by lokI8** [comment]
Outside of those 2 comments it looks good =)

File: quickstack/manifests/controller_common.pp (hunk 1 above)
Comment (genagain) [inline quickstack/manifests/controller_common.pp]: Thanks for the feedback! I just made those changes.

File: quickstack/manifests/compute_common.pp [hunk 2]
```diff
@@ -404,11 +404,27 @@
...
+      }
+    },
+    logging => {
+      'level' => "info"
+    }
+  }
+
+  filebeat::prospector { 'generic':
+      paths => ["/var/log/*.log", "/var/log/secure", "/var/log/messages", "/var/log/ceph/*", "/var/log/nova/*", "/var/log/neutron/*", "/var/log/openvswitch/*"]
+  }
```
Comment (rahulait) [inline quickstack/manifests/compute_common.pp:431]: It seems like you are missing /var/log/cinder/_, /var/log/glance/_, /var/log/horizon/_, /var/log/httpd/_, /var/log/keystone/*

File: quickstack/manifests/compute_common.pp [hunk 3]
```diff
@@ -404,11 +404,27 @@
...
   class {'quickstack::ntp':
     servers => $ntp_local_servers,
   }
 
+  class { 'filebeat':
+    outputs => {
+      'logstash'  => {
+      'hosts'        =>  [$elasticsearch_host],
+      'loadbalance' => true
+      }
```
Comment (rahulait) [inline quickstack/manifests/compute_common.pp:422]: you can indent/remove one space from line 415

File: quickstack/manifests/controller_common.pp [hunk 4]
```diff
@@ -836,5 +836,20 @@
...
+      }
+    },
+    logging => {
+      'level' => "info"
+    }
+  }
 
+   filebeat::prospector { 'generic':
+      paths => ["/var/log/*.log", "/var/log/secure", "/var/log/messages", "/var/log/ceph/*", "/var/log/nova/*", "/var/log/neutron/*", "/var/log/openvswitch/*"]
+    }
```
Comment (rahulait) [inline quickstack/manifests/controller_common.pp:859]: Same as above, missing few log-paths here.

**Comment by rahulait** [comment]
Hmm, this pull request contains 5 commits, but this could have been done with 2 commits (one for the new module, and 2nd for the changes). A good practice would be to not to have the unwanted commits such as "resolving the merge conflicts" since they will then show up on the commit history of the main branch. Maybe you can close this pull request, pull the fresh code and submit a new pull request with 2 commits (one for new module and other for using that module after incorporating the comments).

**Comment by genagain** [comment]
Thank you Jon and Rahul for the feedback! Rahul, thank you for the information. I created a new pull request with only the two necessary commits. Here is the pull request https://github.com/CCI-MOC/kilo-puppet/pull/47. I will be closing this pull request now.

**Pull request closed without merging**